fonttools: 4.21.1 -> 4.26.2;  psautohint: 2.3.0 -> 2.3.1;  afdko: 3.5.1 -> 3.7.1; nototools: 0.2.13 -> 0.2.16

sternenseemann opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Update a bunch of packages in the nototools font toolchain now that afdko 3.7.1 has been released for a bit (they took forever to release it and in between we were stuck with 3.5.x (holding other packages back) because it was broken on darwin).

Still outdated:

* fonttools (4.27.x isn't supported by afdko)
* ~~ufonormalizer (afdko's test suite doesn't like the behavior of 0.6.1)~~ (in `master` with workaround for the test suite)
* tqdm
* more?

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

sternenseemann committed:
python3Packages.fonttools: 4.21.1 -> 4.26.2

We can't yet upgrade to fonttools 4.27.* because afdko doesn't support
it yet.

* Refactor to use pytestCheckHook

* No longer necessary to set locale prior to running the tests

* Python 3.6 support will be dropped soon

Change Log entrys:

* https://github.com/fonttools/fonttools/releases/tag/4.22.0

* https://github.com/fonttools/fonttools/releases/tag/4.22.1

* https://github.com/fonttools/fonttools/releases/tag/4.23.0

* https://github.com/fonttools/fonttools/releases/tag/4.23.1

* https://github.com/fonttools/fonttools/releases/tag/4.24.0

* https://github.com/fonttools/fonttools/releases/tag/4.24.1

* https://github.com/fonttools/fonttools/releases/tag/4.24.2

* https://github.com/fonttools/fonttools/releases/tag/4.24.3

* https://github.com/fonttools/fonttools/releases/tag/4.24.4

* https://github.com/fonttools/fonttools/releases/tag/4.25.0

* https://github.com/fonttools/fonttools/releases/tag/4.25.1

* https://github.com/fonttools/fonttools/releases/tag/4.25.2

* https://github.com/fonttools/fonttools/releases/tag/4.26.0

* https://github.com/fonttools/fonttools/releases/tag/4.26.1

* https://github.com/fonttools/fonttools/releases/tag/4.26.2

sternenseemann committed:
python3Packages.psautohint: 2.3.0 -> 2.3.1

https://github.com/adobe-type-tools/psautohint/blob/master/NEWS.md#v231---july-28-2021

sternenseemann committed:
antlr4_9: init at 4.9.2

* Mostly copied from 4.8.nix

* C++ runtime now requires utfcpp which is solved using an
  ExternalProject in 4.9.2. Since this requires network access, we patch
  it out using a patch picked from antlr4's master.

* Change logs:

  * https://github.com/antlr/antlr4/releases/tag/4.9
  * https://github.com/antlr/antlr4/releases/tag/4.9.1
  * https://github.com/antlr/antlr4/releases/tag/4.9.2

sternenseemann committed:
python3Packages.afdko: 3.5.1 -> 3.7.1

sternenseemann committed:
python3Packages.nototools: 0.2.13 -> 0.2.16

sternenseemann committed:
python3Packages.afdko: run full test suite in passthru.tests

Testsuite delta seems to be 25% still (3min vs 4min on my machine…).

SuperSandro2000 commented on pkgs/development/python-modules/afdko/default.nix:
```diff
@@ -2,37 +2,50 @@
-      sha256 = "0p6a485mmzrbfldfbhgfghsypfiad3cabcw7qlw2rh993ivpnibf";
-    })
-    # fix tests for fonttools 4.21.1
-    (fetchpatch {
```
Please remove the unused fetchpatch from inputs.

SuperSandro2000 commented on pkgs/development/python-modules/afdko/default.nix:
```diff
@@ -2,37 +2,50 @@
 
-  nativeBuildInputs = [ setuptools-scm ];
+  preBuild = ''
+    ninja
```
calling ninja directly probably misses some arguments nix would add automatically.

SuperSandro2000 commented on pkgs/development/python-modules/afdko/no-pypi-build-tools.patch:
```diff
@@ -0,0 +1,24 @@
+           setup_requires=[
+               'wheel',
+               'setuptools_scm',
+-              'scikit-build',
+-              'cmake',
+-              'ninja'
```
Please use subsituteInPlace instead.

SuperSandro2000 commented on pkgs/development/tools/parsing/antlr/4.9.nix:
```diff
@@ -0,0 +1,95 @@
+    inherit version;
+
+    src = fetchurl {
+      url ="https://www.antlr.org/download/antlr-${version}-complete.jar";
```
```suggestion
      url = "https://www.antlr.org/download/antlr-${version}-complete.jar";
```

SuperSandro2000 commented on pkgs/development/tools/parsing/antlr/4.9.nix:
```diff
@@ -0,0 +1,95 @@
+      nativeBuildInputs = [ cmake ninja pkg-config ];
+      buildInputs = [ utf8cpp ]
+        ++ lib.optional stdenv.isLinux libuuid
+        ++ lib.optional stdenv.isDarwin darwin.apple_sdk.frameworks.CoreFoundation;
```
```suggestion
        ++ lib.optional stdenv.isDarwin CoreFoundation;
```

SuperSandro2000 commented on pkgs/development/tools/parsing/antlr/4.9.nix:
```diff
@@ -0,0 +1,95 @@
+{ lib, stdenv, fetchurl, jre
+, fetchpatch, fetchFromGitHub, cmake, ninja, pkg-config, libuuid, utf8cpp, darwin }:
```
```suggestion
, fetchpatch, fetchFromGitHub, cmake, ninja, pkg-config, libuuid, utf8cpp, CoreFoundation
}:
```

sternenseemann replied:
That doesn't work?!

SuperSandro2000 commented on pkgs/development/python-modules/fonttools/default.nix:
```diff
@@ -14,37 +14,38 @@
   # all dependencies are optional, but
   # we run the checks with them
   checkInputs = [
     pytest
```
```suggestion
```

sternenseemann committed:
python3Packages.afdko: skip broken test on aarch64

Reported upstream, hopefully harmless:
https://github.com/adobe-type-tools/afdko/issues/1425

sternenseemann commented:
`x86_64-darwin` is fine when disabling `utf8cpp`'s test suite, see <https://github.com/NixOS/nixpkgs/issues/144265>.

sternenseemann commented:
Rebased on master (keeping staging target), simplified `afdko` derivation a bit.

Given that breaking change according to the release stabilization is

> A change that is likely to break downstream.

We may still merge this for 21.11, I think, since `afdko` has a relatively small set of (direct) reverse dependencies.

sternenseemann commented:
Rebased on top of the last `staging-next` iteration and tested the font builds again. Gonna merge tomorrow, we could ponder backporting, but may be unnecessary hassle.
